src/App.js

```diff
@@ -30,14 +30,63 @@ const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [player, setPlayer] = useState(PLAYER_1)
+  const [winner, setWinner] = useState(null)
+  
 
+  const onClickCallback = (updatedSquare) => {
+    const squareList = [];
+    if (winner === null) {
+      squares.forEach((row, i) => {
+        squareList.push([]);
+        row.forEach(square => {
+          if (square.id === updatedSquare.id && square.value === '') {
+            squareList[i].push(updatedSquare); 
+            if (player === PLAYER_1) {
+              setPlayer(PLAYER_2);
+            } else if (player === PLAYER_2) {
+              setPlayer(PLAYER_1);
+            }
+          } else {
+            squareList[i].push(square);
+          }
+        })
+      })
+      setSquares(squareList);
+
+      checkForWinner(squareList);
+    }
+
+  };
+  
   // Wave 2
   // You will need to create a method to change the square 
   //   When it is clicked on.
   //   Then pass it into the squares as a callback
 
 
-  const checkForWinner = () => {
+  const checkForWinner = (squares) => {
+    const flattenarray = [].concat(...squares);
+    
+    const lines = [
```

Comment:
This is a clever way to work with this 2D array. Consider adding some comments so you remember how it's working. You might also consider how you could use loops to find rows and columns without flattening the array.